client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts

```diff
@@ -323,9 +331,22 @@ const useContactQuestioning = (parameters: useContactQuestioningParameters): use
         updateInteractedContact(interactedContact, InteractedContactFields.IDENTIFICATION_TYPE, newIdentificationType);
     };
 
-    const checkForDuplicateIds = (idToCheck: string, interactedContactId: number) => {
-        return allContactedInteractions.findIndex((contact) => contact.identificationNumber === idToCheck
-            && contact.id !== interactedContactId);
+    const checkForSpecificDuplicateIds = (identificationNumberToCheck: string, interactedContactId: number) => {
+        if(Boolean(identificationNumberToCheck)) {
+            return (allContactedInteractions.findIndex((contact) => contact.identificationNumber === identificationNumberToCheck
+                && contact.id !== interactedContactId) === -1);
+        } else {
+            return !Boolean(identificationNumberToCheck)
+        }
+    }
+
+    const checkAllContactsForDuplicateIds = () => {
+        const allIdentificationNumbersToCheck = allContactedInteractions.map((currContact) => {
```

Comment:
maybe filter instead of mapping?

Reply:
i think if nothing will be returned it will return undefined and length will be same?